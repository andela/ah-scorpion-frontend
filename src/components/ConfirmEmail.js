import React from 'react';

const ConfirmEmail = () => (
  <main className="mt-2em">
    <div className="container p-5 signup-container">
      <div className="alert alert-success" style={{ marginTop: '30px' }}>
        <center>
          <strong>Success</strong>
          <br />
          Please confirm your email address to complete the registration.
        </center>
      </div>
    </div>
  </main>
);

export default ConfirmEmail;
