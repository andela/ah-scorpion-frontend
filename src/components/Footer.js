// This is an example of a "dumb" component which is not aware of Redux

import React from 'react';

const Footer = () => (
  <footer className="bg-dark text-white footer">
    <div className="row">
      <div className="col-md-12 mt-3">
        <p className="text-center text-muted">
          Â© Copyright 2018 Authors Haven - All rights reserved.{" "}
        </p>
      </div>
    </div>
  </footer>
);

export default Footer;
