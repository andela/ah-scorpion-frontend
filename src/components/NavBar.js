import React from 'react';

const NavBar = () => (
  <nav className="navbar navbar-expand-md bg-primary navbar-dark fixed-top">
    <div className="container">
      <a className="navbar-brand" href="/">
        <img
          src="https://res.cloudinary.com/dqvk8ugtp/image/upload/v1535993624/Author_sHaven_logo_sxca8t.png"
          alt="Author's Haven Logo"
          style={{ borderRadius: '10px' }}
          height="60"
          width="50"
        />
      </a>

      <button
        className="navbar-toggler navbar-toggler-right"
        type="button"
        data-toggle="collapse"
        data-target="#navbar2SupportedContent"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <div
        className="collapse navbar-collapse text-center justify-content-end"
        id="navbar2SupportedContent"
      />
    </div>
  </nav>
);

export default NavBar;
