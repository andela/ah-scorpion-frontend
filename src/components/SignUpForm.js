import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import SignUpError from './SignUpError';
import SocialLogin from '../containers/SocialLogin';

class SignUpForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: '',
      confPassword: '',
    };

    this.onFieldChange = this.onFieldChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  onFieldChange(event) {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const { onSubmit, verifyPassword } = this.props;
    const { password, confPassword } = this.state;
    if (verifyPassword(password, confPassword)) {
      onSubmit(this.state);
    }
  }

  render() {
    const {
      onError, errorMsg, isFetching, history,
    } = this.props;
    const {
      email, username, password, confPassword,
    } = this.state;
    return (
      <main className="mt-2em">
        <div className="container p-5 signup-container">
          {onError ? <SignUpError errorMsg={errorMsg} /> : null}
          <div className="card  form-bg ptb-1em">
            <div className="card-header text-center form-bg">
              <h2>Sign Up</h2>
            </div>
            <form className="pb-5 pl-5 pr-5 pt-0 form-bg" onSubmit={this.handleSubmit}>
              <div className="text-center">
                <p>You could sign up with...</p>
                <SocialLogin history={history} />
                <hr />
                <p className="mt-2em">Or</p>
              </div>
              <div className="form-group">
                <input
                  type="email"
                  required
                  className="form-control"
                  id="exampleInputEmail1"
                  placeholder="Email Address"
                  name="email"
                  value={email}
                  onChange={this.onFieldChange}
                />
              </div>
              <div className="form-group">
                <input
                  type="text"
                  required
                  className="form-control"
                  id="exampleInputUsername"
                  placeholder="Username"
                  name="username"
                  value={username}
                  onChange={this.onFieldChange}
                />
              </div>
              <div className="form-group">
                <input
                  type="password"
                  required
                  className="form-control"
                  id="exampleInputPassword1"
                  placeholder="Password"
                  name="password"
                  value={password}
                  onChange={this.onFieldChange}
                />
              </div>
              <div className="form-group">
                <input
                  type="password"
                  required
                  className="form-control"
                  id="exampleInputPassword2"
                  placeholder="Confirm Password"
                  name="confPassword"
                  value={confPassword}
                  onChange={this.onFieldChange}
                />
              </div>
              <div className="text-center">
                <button
                  type="submit"
                  id="submitButton"
                  className="btn btn-primary"
                  disabled={isFetching}
                >
                  {isFetching ? 'Loading...' : 'Sign Up'}
                </button>
                <p>
                  Already have an account? Login
                  <Link to="/login"> here</Link>
                </p>
              </div>
            </form>
          </div>
        </div>
      </main>
    );
  }
}

SignUpForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onError: PropTypes.bool.isRequired,
  isFetching: PropTypes.bool.isRequired,
  errorMsg: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.string),
    PropTypes.object,
    PropTypes.string,
  ]),
  verifyPassword: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

SignUpForm.defaultProps = {
  errorMsg: null,
};

export default SignUpForm;
