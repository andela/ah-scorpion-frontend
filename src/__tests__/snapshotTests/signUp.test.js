import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';

import NavBar from '../../components/NavBar';
import Footer from '../../components/Footer';
import ConfirmEmail from '../../components/ConfirmEmail';
import SignUpError from '../../components/SignUpError';
import SignUpForm from '../../components/SignUpForm';

describe('The signUp view', () => {
  describe('The NavBar component', () => {
    it('Should not regress', () => {
      const tree = renderer.create(
        <MemoryRouter>
          <NavBar />
        </MemoryRouter>,
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('The Footer component', () => {
    it('Should not regress', () => {
      const tree = renderer.create(
        <MemoryRouter>
          <Footer />
        </MemoryRouter>,
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('The ConfirmEmail component', () => {
    it('Should not regress', () => {
      const tree = renderer.create(
        <MemoryRouter>
          <ConfirmEmail />
        </MemoryRouter>,
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('The SignUpError component', () => {
    it('Should not regress', () => {
      const tree = renderer.create(
        <MemoryRouter>
          <SignUpError errorMsg="An error occurred" />
        </MemoryRouter>,
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('The SignUpForm component', () => {
    const testProps = {
      onSubmit: () => 1,
      verifyPassword: () => 1,
      history: { push: () => 1 },
      errorMsg: 'An error occurred',
    };
    it('Should initially render without an error or loading status', () => {
      const tree = shallow(
        <MemoryRouter>
          <SignUpForm {...testProps} onError={false} isFetching={false} />
        </MemoryRouter>,
      );
      // expect(shallowToJSON(tree)).toMatchSnapshot();
    });
    it('Should show loading status when request is pending', () => {
      const tree = shallow(
        <MemoryRouter>
          <SignUpForm {...testProps} onError={false} isFetching />
        </MemoryRouter>,
      );
      // expect(shallowToJSON(tree)).toMatchSnapshot();
    });
    it('Should show the error message when request fails', () => {
      const tree = shallow(
        <MemoryRouter>
          <SignUpForm {...testProps} onError isFetching={false} />
        </MemoryRouter>,
      );
      // expect(shallowToJSON(tree)).toMatchSnapshot();
    });
  });
});
