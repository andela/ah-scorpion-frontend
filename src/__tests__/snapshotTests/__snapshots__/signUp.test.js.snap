// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The signUp view The ConfirmEmail component Should not regress 1`] = `
<main
  className="mt-2em"
>
  <div
    className="container p-5 signup-container"
  >
    <div
      className="alert alert-success"
      style={
        Object {
          "marginTop": "30px",
        }
      }
    >
      <center>
        <strong>
          Success
        </strong>
        <br />
        Please confirm your email address to complete the registration.
      </center>
    </div>
  </div>
</main>
`;

exports[`The signUp view The Footer component Should not regress 1`] = `
<footer
  className="bg-dark text-white footer"
>
  <div
    className="row"
  >
    <div
      className="col-md-12 mt-3"
    >
      <p
        className="text-center text-muted"
      >
        Â© Copyright 2018 Authors Haven - All rights reserved.
      </p>
    </div>
  </div>
</footer>
`;

exports[`The signUp view The NavBar component Should not regress 1`] = `
<nav
  className="navbar navbar-expand-md bg-primary navbar-dark fixed-top"
>
  <div
    className="container"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div
        className="navbar-brand"
      >
        <b>
          Authors Haven
        </b>
      </div>
    </a>
    <button
      className="navbar-toggler navbar-toggler-right"
      data-target="#navbar2SupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse text-center justify-content-end"
      id="navbar2SupportedContent"
    />
  </div>
</nav>
`;

exports[`The signUp view The SignUpError component Should not regress 1`] = `
<div
  className="alert alert-danger"
>
  <center>
    An error occurred
  </center>
</div>
`;
<<<<<<< HEAD

exports[`The signUp view The SignUpForm component Should initially render without an error or loading status 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <SignUpForm
    errorMsg="An error occurred"
    history={
      Object {
        "push": [Function],
      }
    }
    isFetching={false}
    onError={false}
    onSubmit={[Function]}
    verifyPassword={[Function]}
  />
</Router>
`;

exports[`The signUp view The SignUpForm component Should show loading status when request is pending 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <SignUpForm
    errorMsg="An error occurred"
    history={
      Object {
        "push": [Function],
      }
    }
    isFetching={true}
    onError={false}
    onSubmit={[Function]}
    verifyPassword={[Function]}
  />
</Router>
`;

exports[`The signUp view The SignUpForm component Should show the error message when request fails 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <SignUpForm
    errorMsg="An error occurred"
    history={
      Object {
        "push": [Function],
      }
    }
    isFetching={false}
    onError={true}
    onSubmit={[Function]}
    verifyPassword={[Function]}
  />
</Router>
`;
=======
>>>>>>> fix bug for user profile
